modules/packages/LinearAlgebra.chpl

```diff
@@ -938,6 +938,125 @@ proc trace(A: [?D] ?eltType) {
 }
 
 
+/* Perform a Cholesky factorization on matrix ``A``.  ``A`` must be square.
+   Argument ``lower`` indicates whether to return the lower or upper
+   triangular factor.  Matrix ``A`` is not modified.
+ */
+proc cholesky(A: [] ?t, lower = true) where A.rank == 2 &&
+                                            (isRealType(t) ||
+                                             isComplexType(t)) {
+  assert(A.domain.dim(1) == A.domain.dim(2));
+  var copy = A;
+  const uploStr = if lower then "L" else "U";
+  potrf(lapack_memory_order.row_major, uploStr, copy);
+
+  // tril and triu make/return an extra copy.  Should we zero the unused
+  // triangle of the array manually instead to avoid the copy?
+  return if lower then tril(copy) else triu(copy);
+}
+
+
+/* Find the eigenvalues of matrix ``A``. ``A`` must be square.
+
+   * If ``left`` is ``true`` then the "left" eigenvectors are computed. The
+     return value is a tuple with two elements:
+     ``(eigenvalues, leftEigenvectors)``
+
+   * If ``right`` is ``true`` then the "right" eigenvectors are computed. The
+     return value is a tuple with two elements:
+     ``(eigenvalues, rightEigenvectors)``
+
+   * If ``left`` and ``right`` are both ``true`` then both eigenvectors are
+     computed.  The return value is a tuple with three elements:
+     ``(eigenvalues, leftEigenvectors, rightEigenvectors)``
+
+   * If ``left`` and ``right`` are both ``false`` only the eigenvalues are
+     computed, and returned as a single array.
+
+ */
+proc eigvals(A: [] ?t, param left = false, param right = false)
```

Comment:
Just occurred to me: Would it be more Chapelistic to camelCase this as `eigVals()`?

Reply:
It's short for the single word "eigenvalues", so IMO, no.